platform/src/Panel.js

```diff
@@ -97,27 +97,14 @@ class Panel {
         return this.type;
     }
 
-    buttonHtml(icon, hint) {
-        return "<span class='mif-" + icon + "' data-role='hint' data-hint-text='" + hint + "' data-hint-position='bottom'></span>";
-    }
-
-
     /**
      * Add the buttons to the page
-     * @param {object[]} buttons  Objects with attributes: icon, hint, action
-     * 
-     * TODO Support image files for icon
+     * @param {Button[]} buttons  the Button objects to add
      */
     addButtons(buttons){
 
         var buttonViewData= buttons.map( (btn) => {
-            var buttonData={};
-            
-            buttonData.html= this.buttonHtml(btn.icon, btn.hint);
-            buttonData.cls= "sys-button";
-            buttonData.onclick= btn.action;
-
-            return buttonData;
+            return btn.getView();
```

Comment:
But elsewhere you're already calling `.getView()` when defining the buttons. Are these consistent?

Reply:
There are two ways Buttons are intialised
- for built-in panels that aren't customisable they are intialised in the panel constructor with a call to the panel's getButtons() method e.g. the ConsolePanel line 11.

- for panels that are customisable like those defined by tools in the configuration they use the Panel super class'  addButtons() method called during platform intialisation in Playground.js line 352.

With the added customisation of panels that override those defined in tool panel def there are likely to be odd behaviours introduced e.g. overriding a non-customisable panel's button.

Reply:
But why are those uses not consistent? One uses `getView()` in the definition of the button, the other uses it when adding the button via `addButtons()`.

How would buttons be overwritten and what would happen in that case?

Reply:
I agree that the addButton() can be used in same way for the built-in panels so the getButtons() can be easily removed.

For overwriting the built-in panels buttons its likely you would be able to define buttons new buttons but not reference the panel defined ones by id. I will test this case though to see.

Reply:
For the case of trying to overwrite a built-in panel's buttons the the panels config would have no buttons declared since they are instantiated by the panel's js. Therefore panelDefinition.buttons is undefined so the buttons are not modified by lines 335 or 352 - this is the intended behaviour confirmed by test.